src/v3/src/components/Widget/index.tsx

```diff
@@ -143,8 +146,19 @@ export const Widget: FunctionComponent<WidgetProps> = (widgetProps) => {
         return;
       }
       const transaction = await authClient.idx.start({
-        stateHandle: stateToken,
+        stateHandle,
       });
```

Comment:
Sorry, one more question, what happens if this `start` function throws an exception because say the stateHandle is expired? Will we automatically retry the bootstrap? If they refresh will it use the stateHandle from widgetProps instead of the one stored in session?

Reply:
Good point.
Moved restart logic   in catch block as it was done in v2.
Then `stateHandle` would be cleared from session and  bootstrap will be restarted with `stateHandle` from `widgetProps` if any error was thrown or `requestDidSucceed` is false or there is an error message.